webflow-hacks/form-hacks/29/29.ts

```diff
@@ -0,0 +1,22 @@
+// when DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  // declare constant selectors
+  const FORM_SELECTOR = '[fs-hacks-element="hack29-form"]';
+  const NAME_INPUT_SELECTOR = '[fs-hacks-element="hack29-name-input"]';
+  const MESSAGE_SELECTOR = '.w-form-done';
+  const form = document.querySelector<HTMLFormElement>(FORM_SELECTOR);
+  // early return
+  if (!form) return;
+  // when form is submitted
+  form.addEventListener('submit', function () {
+    const nameInput = this.querySelector<HTMLInputElement>(NAME_INPUT_SELECTOR);
+    const messageDiv = this.querySelector<HTMLDivElement>(MESSAGE_SELECTOR);
```

Comment:
There's no reason to have those selectors inside the event listener, I think it's better to put them outside, right after the early return when the form is null.